plugins/GSdx/GSTextureCache.cpp

```diff
@@ -268,25 +270,16 @@ GSTextureCache::Source* GSTextureCache::LookupSource(const GIFRegTEX0& TEX0, con
 				uint32 t_psm = (t->m_dirty_alpha) ? t->m_TEX0.PSM & ~0x1 : t->m_TEX0.PSM;
 
 				if (GSUtil::HasSharedBits(bp, psm, t->m_TEX0.TBP0, t_psm)) {
-					if (!s_IS_OPENGL && (psm == PSM_PSMT8)) {
-						// OpenGL can convert the texture directly in the GPU. Not sure we want to keep this
-						// code for DX. It fixes effect but it is slow (MGS3)
-
-						// It is a complex to convert the code in shader. As a reference, let's do it on the CPU, it will
-						// be slow but
-						// 1/ it just works :)
-						// 2/ even with upscaling
-						// 3/ for both DX and OpenGL
-
-						// Gregory: to avoid a massive slow down for nothing, let's only enable
-						// this code when CRC is below the FULL level
-						if (m_crc_hack_level < CRCHackLevel::Full)
-							Read(t, t->m_valid);
-						else
-							dst = t;
-					} else {
+					// It is a complex to convert the code in shader. As a reference, let's do it on the CPU, it will be slow but
+					// 1/ it just works :)
+					// 2/ even with upscaling
+					// 3/ for both DX and OpenGL
+					if (m_cpu_8bit_fb_coversion && psm == PSM_PSMT8)
```

Comment:
I wonder whether it would be better to have the option work the other way around (slower but better rendering by default, with the hack option to enable faster but worse rendering).

Reply:
It's probably a bad idea if it causes slowdowns and people dunno what's causing the issue.

Reply:
Unfortunately the impact on the perf could be huge. Read will trigger a synchronous read from the GPU memory. Which mean
1/ flush the driver thread
2/ flush all GPU commands
3/ finally read GPU memory

If the GPU is late on the rendering, the delay could be 1 or 2 renderer frames.